pysr/sr.py

```diff
@@ -2586,3 +2583,122 @@ def _mutate_parameter(param_name: str, param_value):
         return False
 
     return param_value
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        self.recursive_history_length = recursive_history_length
+        super().__init__(**kwargs)
+
+    def fit(
+        self,
+        X,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+    ) -> "PySRRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_times, ...).
+            Multidimensional time series data is supported, but the more dimensions
+            provided, the worse the regressor will perform.
+        weights : ndarray | pandas.DataFrame
+            Weight array of the same shape as `X`, but not for the
+            first recurrence_history_length terms. Therefore, the shape is
+            (n_samples-recurrence_history_length, 1) or (1, n_samples-recurrence_history_length)
+            Each element is how to weight the mean-square-error loss
+            for that particular element of `X`. Alternatively,
+            if a custom `loss` was set, it will can be used
+            in arbitrary ways.
+        variable_names : list[str]
+            A list of names for the variables, rather than "xt_1", "xt_2", etc.
```

Comment:
I think the variable names passed here should be for each feature. i.e., this should be equal to shape `(n_features,)`. Then the internal method should automatically turn that into `(n_features * recurrence_history_length,)`.